sympy/solvers/solveset.py

```diff
@@ -1121,3 +1126,625 @@ def linsolve(system, *symbols):
+    .. math:: 4*x**2 + y**2 - 5 = 0
+
+    `system  = [x*y - 1, 4*x**2 + y**2 - 5]`
+    `symbols = [x, y]`
+
+    Examples
+    ========
+
+    >>> from sympy.core.symbol import symbols
+    >>> from sympy.solvers.solveset import nonlinsolve
+    >>> x, y, z = symbols('x, y, z', real=True)
+    >>> nonlinsolve([x*y - 1, 4*x**2 + y**2 - 5], [x, y])
+    {(-1, -1), (-1/2, -2), (1/2, 2), (1, 1)}
+
+    * Positive dimensional system and complements:
+
+    >>> from sympy import pprint
+    >>> from sympy.polys.polytools import is_zero_dimensional
+    >>> a, b, c, d = symbols('a, b, c, d', real=True)
+    >>> foo =  a + b + c + d
+    >>> bar = a*b + b*c + c*d + d*a
+    >>> foo_bar = a*b*c + b*c*d + c*d*a + d*a*b
+    >>> bar_foo = a*b*c*d -1
+    >>> system = [foo, bar, foo_bar, bar_foo]
+    >>> is_zero_dimensional(system)
+    False
+    >>> pprint(nonlinsolve(system, [a, b, c, d]), use_unicode=False)
+      -1       1
+    {(---, -d, -, {d} \ {0})}
+       d       d
+    >>> nonlinsolve([(x+y)**2 - 4, x + y - 2], [x, y])
+    {(-y + 2, y)}
+
+    * If Non linear system having non polynomial equation have complex
+    solution for particular symbol and no real solution, then `nonlinsolve`
+    returns it's general solution.
+
+    >>> from sympy import exp, sin
+    >>> nonlinsolve([exp(x) - sin(y), y**2 - 4], [x, y])
+    {(log(sin(2)), 2), (ImageSet(Lambda(_n, I*(2*_n*pi + pi) +
+        log(sin(2))), Integers()), -2), (ImageSet(Lambda(_n, 2*_n*I*pi +
+        Mod(log(sin(2)), 2*I*pi)), Integers()), 2)}
+
+
+    * If Non linear system have all the equations polynomial, then it
+    returns both real and complex solutions.
+
+    >>> from sympy import sqrt
+    >>> nonlinsolve([x**2 - 2*y**2 -2, x*y - 2], [x, y])
+    {(-2, -1), (2, 1), (-sqrt(2)*I, sqrt(2)*I), (sqrt(2)*I, -sqrt(2)*I)}
+
+
+    * If system is linear and positive dimensional system, `nonlinsolve` can
+    solve this system (because it is using `groebner` function to get the
+    groebner basis and then `substitution` function basis as system). But it is
+    not recommended to solve linear system using `nonlinsolve`, because
+    `linsolve` is better for all kind of linear system.
+
+    >>> nonlinsolve([x + 2*y -z - 3, x - y - 4*z + 9 , y + z - 4], [x, y, z])
+    {(3*z - 5, -z + 4, z)}
+
+    * system having polynomial equations:
+
+    >>> e1 = sqrt(x**2 + y**2) - 10
+    >>> e2 = sqrt(y**2 + (-x + 10)**2) - 3
+    >>> nonlinsolve((e1, e2), (x, y))
+    {(191/20, -3*sqrt(391)/20), (191/20, 3*sqrt(391)/20)}
+    >>> nonlinsolve([x**2 + 2/y - 2, x + y - 3], [x, y])
+    {(1, 2), (1 + sqrt(5), -sqrt(5) + 2), (-sqrt(5) + 1, 2 + sqrt(5))}
+    >>> nonlinsolve([x**2 + 2/y - 2, x + y - 3], [y, x])
+    {(2, 1), (2 + sqrt(5), -sqrt(5) + 1), (-sqrt(5) + 2, 1 + sqrt(5))}
+
+    How nonlinsolve is better than old solver `_solve_system` :
+    ===========================================================
+
+    * A positive dimensional system solver : nonlinsolve can return
```

Comment:
>  I hope it can.

I think we shouldn't generalise that `nonlinsolve` can solve a positive dimensional system based on a single example. I am pretty sure that simple googling would result in several examples of a positive dimensional system.